dev/TEAL.md

```diff
@@ -358,6 +358,8 @@ A program starts with a varuint declaring the version of the compiled code. Any
 
 For version 1, subsequent bytes after the varuint are program opcode bytes. Future versions could put other metadata following the version identifier.
 
+It is important to prevent newly-introduced transaction fields from breaking assumptions made by older versions of TEAL. If one of the transactions in a group will execute a TEAL program whose version predates a given field, that field must not be set anywhere in the transaction group, or the group will be rejected. For example, executing a TEAL version 1 program on a transaction with RekeyTo set to a nonzero address will cause the program to fail, regardless of the other contents of the program itself.
+
```

Comment:
We should also describe the way this constraint is codified, something like the following text:

----------
This requirement is enforced as follows:
  
* For every transaction in a group, compute the earliest version of the protocol that this transaction complies with. For example a transaction with a nonzero RekeyTo field will have version (at least) 2.
  
* Compute the largest version number across all the transactions in the group, call it `maxVerNo`

* Any TEAL program in this group with version smaller than `maxVerNo` will fail.